### DISCUSSION TITLE
Hulls with a small hole between their borders (but have a connecting gap) will get the character wet and cause drowning.

### DISCUSSION THREAD

**Discussion by NilanthAnimosus** [FakeFishGames/Barotrauma#14107] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

If hulls do not touch or overlap when having a gap between them, although the gap works correctly there are small issues with a character getting wet and losing air inside the small distance between the hull borders.

![image](https://github.com/FakeFishGames/Barotrauma/assets/29177976/1325b855-fdb5-46e7-aed2-e5d9825ae23d)

Pressure issues (by teleporting outside when corssing the gap) were fixed by (Link to private repo) PR: https://github.com/FakeFishGames/Barotrauma-development/issues/5423 but these smaller issues still remain. Has become more visible due to the minimum hallway distance being removed from outposts.

### Reproduction steps

1. Create two hulls with a small hole between two hull borders (2-20 pixels) that is still within a Gap connecting them (IE. door or placed gap).
2. Find the character gets wet when running through the gap.
3. Stand in the distance between hulls and find the character suffocates but is not crushed or triggering items that get "wet".

### Bug prevalence

Happens regularly

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.5.4.0 (unstable)

### Which operating system did you encounter this bug on?

Windows

**Comment by G-M-twostay** [comment]
shouldn't this be expected behavior since the character is technically standing in open ocean as anything outside of hull is considered open ocean?

**Comment by NilanthAnimosus** [comment]
Yes and no, its more of an issue with the generation of outposts, which might cause this to occur, so theres some code in place that (without looking at the code itself, but more explaining it) when crossing those small holes that have a gap (so their connected hulls) you don't get crushed by pressure by "Exiting" the submarine as its not intended for say, a gap between two hulls not exposed to the ocean to kill you between them as you cross (such as through a door).

its already been patched to prevent the main issue, but this is just a small lingering visual issue (plus the loss of oxygen if you stand there) more or less.
